File: ubbe/app/src/main/java/com/example/ubbe/MainActivity.java
package com.example.ubbe;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import com.example.ubbe.adapter.ProductListAdapter;
import com.example.ubbe.data.ProductData;
import com.example.ubbe.model.Product;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    private RecyclerView rvProduct;
    private ArrayList<Product> list = new ArrayList<>();

    // Todo: make image fit to container

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        rvProduct = findViewById(R.id.rvProduct);
        rvProduct.setHasFixedSize(true);

        list.addAll(ProductData.getListData());
        showRecycleCardView();
    }

    private void showRecycleCardView() {
        rvProduct.setLayoutManager(new LinearLayoutManager(this));
        ProductListAdapter productListAdapter = new ProductListAdapter(MainActivity.this, list);
        rvProduct.setAdapter(productListAdapter);

//        productListAdapter.setListener(new ProductListAdapter.Listener() {
//            @Override
//            public void onClick(int position) {
//                Intent intent = new Intent(MainActivity.this, ProductDetail.class);
//                intent.putExtra(ProductDetail.EXTRA_PRODUCT_ID, position);
//                startActivity(intent);
//                Log.d("position", "onClick: " + position);
//            }
//        });
    }

}

File: ShopApp/app/src/main/java/com/example/shopapp/MainActivity.kt
package com.example.shopapp

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class MainActivity : AppCompatActivity(), View.OnClickListener {
    private lateinit var rvProduct: RecyclerView
    private var list: ArrayList<Product> = arrayListOf()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        rvProduct = findViewById(R.id.rvProducts)
        rvProduct.setHasFixedSize(true)

        list.addAll(ProductData.listData)
        showRecyclerList()

        val btnDetailProduct : androidx.cardview.widget.CardView = findViewById(R.id.list_item)
        btnDetailProduct.setOnClickListener(this)
    }

    private fun showRecyclerList() {
        rvProduct.layoutManager = LinearLayoutManager(this)
        val productListAdapter = ProductListAdapter(list)
        rvProduct.adapter = productListAdapter
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.list_item -> {
                val moveIntent = Intent(this@MainActivity, DetailProduct::class.java)
                startActivity(moveIntent)
            }
        }
    }
}